src/Consolonia.Core/Text/Fonts/AsciiArtTypeface.cs

```diff
@@ -0,0 +1,405 @@
+                move = Math.Min(move, rightLine.Length);
+
+                // Track the minimum move across all lines (tightest constraint)
+                if (move < minMove)
+                    minMove = move;
+            }
+
+            // If we never found any valid line, return 0
+            if (minMove == int.MaxValue)
+                return 0;
+
+            return Math.Max(0, minMove);
+        }
+
+        /// <summary>
+        ///     Check if two characters can smush together, returning the resulting character or '\0' if they can't
+        /// </summary>
+        private char TrySmush(char leftChar, char rightChar)
+        {
+            // If either is space, they can't smush (but can kern)
+            if (leftChar == ' ' || rightChar == ' ')
+                return '\0';
+
+            // Hardblank is treated specially
+            if (leftChar == Hardblank || rightChar == Hardblank)
+            {
+                if (LayoutMode.HasFlag(LayoutMode.Hardblank))
+                    return leftChar == Hardblank ? rightChar : leftChar;
+                return '\0';
+            }
+
+            // Check smush rules
+            if (LayoutMode.HasFlag(LayoutMode.Equal))
+                if (leftChar == rightChar)
+                    return leftChar;
+
+            if (LayoutMode.HasFlag(LayoutMode.Lowline))
+            {
+                if (leftChar == '_' && "|/\\[]{}()<>".Contains(rightChar))
+                    return rightChar;
+                if (rightChar == '_' && "|/\\[]{}()<>".Contains(leftChar))
+                    return leftChar;
+            }
+
+            if (LayoutMode.HasFlag(LayoutMode.Hierarchy))
+            {
+                string hierarchy = "|/\\[]{}()<>";
+                int leftLevel = hierarchy.IndexOf(leftChar);
+                int rightLevel = hierarchy.IndexOf(rightChar);
+
+                if (leftLevel >= 0 && rightLevel >= 0)
+                    return leftLevel > rightLevel ? leftChar : rightChar;
+            }
+
+            if (LayoutMode.HasFlag(LayoutMode.Pair))
+            {
+                if (leftChar == '[' && rightChar == ']' || leftChar == ']' && rightChar == '[')
+                    return '|';
+                if (leftChar == '{' && rightChar == '}' || leftChar == '}' && rightChar == '{')
+                    return '|';
+                if (leftChar == '(' && rightChar == ')' || leftChar == ')' && rightChar == '(')
+                    return '|';
+            }
+
+            if (LayoutMode.HasFlag(LayoutMode.BigX))
+            {
+                if (leftChar == '/' && rightChar == '\\')
+                    return '|';
+                if (leftChar == '\\' && rightChar == '/')
+                    return 'Y';
+                if (leftChar == '>' && rightChar == '<')
+                    return 'X';
+            }
+
+            return '\0';
+        }
+
+        // Resharper disable AssignNullToNotNullAttribute
```

Comment:
There is no null assigment in this method

Reply:
the linter was totally giving me shit and I couldn't figure out what it was talking about. I have tried removing again, we'll see